guides/source/7_2_release_notes.md

```diff
@@ -21,6 +26,251 @@ guide.
+
+Rails 7.2 adds the ability to generate a development container configuration for your application. This configuration
+includes a `.devcontainer` folder with a `Dockerfile`, a `docker-compose.yml` file, and a `devcontainer.json` file.
+
+By default the dev container contains the following:
+
+* A redis container for use with Kredis, Action Cable etc.
+* A database (SQLite, Postgres, MySQL or MariaDB)
+* A Headless chrome container for system tests
+* Active Storage configured to use the local disk and with preview features working
+
+To generate a new application with a development container, you can run:
+
+```bash
+rails new myapp --devcontainer
+```
+
+For existing applications, a `devcontainer` command is now available:
+
+```bash
+rails devcontainer
+```
+
+For more information, see the [Getting Started with Dev Containers](https://edgeguides.rubyonrails.org/getting_started_with_devcontainer.html) guide.
+
+### Add browser version guard by default
+
+Rails now adds the ability to specify the browser versions that will be allowed to access all actions
+(or some, as limited by `only:` or `except:`).
+
+Only browsers matched in the hash or named set passed to `versions:` will be blocked if they're below the versions
+specified.
+
+This means that all other unknown browsers, as well as agents that aren't reporting a user-agent header, will be allowed access.
+
+A browser that's blocked will by default be served the file in `public/406-unsupported-browser.html` with a HTTP status
+code of "406 Not Acceptable".
+
+Examples:
+
+```ruby
+class ApplicationController < ActionController::Base
+  # Allow only browsers natively supporting webp images, web push, badges, import maps, CSS nesting + :has
+  allow_browser versions: :modern
+end
+
+class ApplicationController < ActionController::Base
+  # All versions of Chrome and Opera will be allowed, but no versions of "internet explorer" (ie). Safari needs to be 16.4+ and Firefox 121+.
+  allow_browser versions: { safari: 16.4, firefox: 121, ie: false }
+end
+
+class MessagesController < ApplicationController
+  # In addition to the browsers blocked by ApplicationController, also block Opera below 104 and Chrome below 119 for the show action.
+  allow_browser versions: { opera: 104, chrome: 119 }, only: :show
+end
+```
+
+For more information, see the [allow_browser](https://api.rubyonrails.org/classes/ActionController/AllowBrowser/ClassMethods.html#method-i-allow_browser)
+documentation.
+
+### Make Ruby 3.1 the new minimum version
+
+Until now, Rails only droped compatibility with older Rubies on new majors version.
+We are changing this policy because it causes us to keep compatibility with long
+unsupported versions of Ruby or to bump the Rails major version more often, and to
+drop multiple Ruby versions at once when we bump the major.
+
+Now we are going to drop Ruby versions on minor versions, following the recommendation
+from th Ruby team at the time of the release.
```

Comment:
```suggestion
We will now drop Ruby versions that are end-of-life on minor Rails versions at the time of the release.
```

Maybe we could clarify that we'll drop all EOL rubies when releasing a new minor? Would be more clear I think.